src/util/evalUnits.ts

```diff
@@ -53,19 +81,26 @@ export enum EvalMessages {
  * the first is the actual calculated number (or null if none) while
  * the second is the message that should appear in case of error
  */
-export function evalUnits(input: string): [number | null, string] {
+export function evalUnits(input: string): [BN | number | null, string] {
   if (!floats.test(input) && !ints.test(input) && !alphaInts.test(input) && !alphaFloats.test(input)) {
     return [null, EvalMessages.GIBBERISH];
   }
-
   if (floats.test(input) || ints.test(input)) {
-    return [parseFloat(input.replace(/[,]/g, '.')), EvalMessages.SUCCESS];
-  } else if (alphaInts.test(input) || alphaFloats.test(input)) {
-    const numericPart = parseFloat(input.replace(/[,]/g, '.'));
-    const charPart = input.replace(/[0-9.,]/g, '');
-    const siVal = si.find((s) => s.symbol === charPart);
-    return siVal ? [numericPart * siVal.value, EvalMessages.SUCCESS] : [null, EvalMessages.SYMBOL_ERROR];
-  } else {
+    const result = parseFloat(input.replace(/[,]/g, '.'));
+    return [result, EvalMessages.SUCCESS];
+  }
+  if (alphaInts.test(input) || alphaFloats.test(input)) {
+    const containDecimal = !input.replace(/[,]/g, '.').includes('.');
+    return transformShorthandToBN(input, containDecimal);
+  }
+  if (new BN(input).gte(new BN(Number.MAX_SAFE_INTEGER))) {
```

Comment:
What's the purpose of thi scheck? The point of using `BN` is to exactly support numbers larger than `MAX_SAFE_INTEGER`.